contributor-book/src/guides/adding-a-new-operation-to-burn.md

```diff
@@ -0,0 +1,216 @@
+# Adding a new operation to burn
+
+Let's discuss how one might go about adding new operators to Burn, using the example of the
+pow operator added in [this PR](https://github.com/tracel-ai/burn/pull/1133/files). In that
+PR, the following things took place (albeit not in this order)
+
+## Adding the Op to burn-tensor
+
+`burn-tensor` is the crate that defines all tensor operations that need to be implemented by the
+various backends. The core of this lies in `burn-tensor/src/tensor/api/numeric.rs`, which is home to
+the numeric trait and its implementation for the different tensor types. The numeric trait is the
+home of all tensor operations that are numeric in nature and that are shared by `Int` and `Float`
+Tensor types. More information on the relationship between Tensor modules can be found under the
+section for [Tensor Architecture](../project-architecture/Tensor.md#tensorops).
+
+Here is where pow was added to `burn-tensor/src/tensor/api/numeric.rs`:
+
+1. for the
+   [`Tensor<Backend,Dimension,Kind>` struct](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L553)
+2. for the
+   [numeric trait](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L1618)
+3. for the implementation of numeric for
+   [float](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L2186)
+   and
+   [int](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L1903)
+
+Tensor is a struct that has a single member: `primitive` (defined
+[here](https://github.com/tracel-ai/burn/blob/main/burn-tensor/src/tensor/api/base.rs)), that is
+defined by it's
+[`Kind`](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/kind.rs#L15):
+one of `Bool`, `Float`, or `Int` (those linked in 3). These call the ops for that data type defined
+in the
+[`Backend`](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/backend/base.rs#L52)
+supertrait[^1]. This is the trait that is then implemented by the different `burn-` backends (such
+as `burn-ndarray` and `burn-wgpu`) which implement the functions if no default is provided.
+
+In this case, we only need to worry about `Bool` Tensors. Ops for `Float` is implemented under
```

Comment:
I think we don't have to worry about Bool 😅

Reply:
yeah bools are pretty sure of themselves. Let me fix this